lib/queue.rb

```diff
@@ -1,31 +1,49 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(20)
+    @start = 0
+    # @end = 0
+    @size = 0
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if size < 20
+      @store[(@start + @size) % 20] = element
+      @size += 1
+    end
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    if @size != 0
+      element = @store[@start] 
+      @start = (@start + 1 ) % 20
```

Comment:
Avoid hardcoding the `20`

```suggestion
      @start = (@start + 1 ) % @store.length
```